### ISSUE TITLE
JWTRule support specify jwt requirement

### ISSUE THREAD

**Issue by wulianglongrd** [istio/istio#43982]
Currently, when jwt is used for authentication, the verification will pass when jwt is valid or not provided. This requirement does not apply in some scenarios:

1. During the process of adding authentication capabilities to existing services, or when the strategy needs to be adjusted for some reason, it is hoped to temporarily tolerate authentication errors (use logs and metrics to confirm errors), so as to ensure that unexpected configuration errors will not cause impaired traffic in the production environment (*not supported*)

2. In some scenarios, we only use authentication without authorization, and we want to strictly verify the jwt (must be provided and valid) (*authn is not directly supported, but authz can be used to support it*)

Therefore, we need a new API to control the following three requirements of jwt:
1. Allow jwt not provided, but reject when jwt is provided but validation fails (default behavior, already supported)
2. Always allow, even if jwt is not provided or validation fails (*not supported*)
3. Jwt must be provided and validation successfully (authz can indirectly support)

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[] Multi Cluster
[] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by wulianglongrd** [comment]
currently, valid and empty tokens are accepted, but invalid tokens are rejected

**Comment by ramaraochavali** [comment]
@wulianglongrd Is this implemented? I think it makes sense to have the API proposed with additional enum value that also says DONOT_ALLOW_IF_MISSING or some thing so that Jwt filter strictly validates the presence of JWT. Curious, why you closed this?

**Comment by wulianglongrd** [comment]
> @wulianglongrd Is this implemented? I think it makes sense to have the API proposed with additional enum value that also says DONOT_ALLOW_IF_MISSING or some thing so that Jwt filter strictly validates the presence of JWT. Curious, why you closed this?

Thanks for the attention, this issue has been created for over a month, but it seems that the community is not interested in it

**Comment by ramaraochavali** [comment]
Sorry. I missed this. We hare interested in this feature as well. Our use case is we want to enforce jwt validation (not skip if it is missing or failed etc). Can you please reopen the API PR and add this support also?

**Comment by wulianglongrd** [comment]
I have re-opened, please take a look, thanks.

**Comment by hexiaodai** [comment]
We have the same problem.

**Comment by wulianglongrd** [comment]
Can you describe your usage scenario and what problem you want to solve? Thanks.

**Comment by hanxiaop** [comment]
> Can you describe your usage scenario and what problem you want to solve? Thanks.

@wulianglongrd We originally have a RA to validate jwts. Currently, we want to allow some requests with jwts not from that issuer to bypass the RA and instead have them validated in services or by authz. However, from the current configured `allow_missing` rules, if a jwt is provided, it will be rejected by the RA before our validation logics.

**Comment by hanxiaop** [comment]
@wulianglongrd Thanks for the info. The token that causes the failure cannot be validated by adding another JWT rules, and that token is required for our services. I also think tolerating the failure is reasonable.

**Comment by louiscryan** [comment]
We likely need to support

https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/jwt_authn_filter#requirementrule

**Comment by wulianglongrd** [comment]
@louiscryan Yes, we have submitted a PR https://github.com/istio/api/pull/2733, but it has not been merged yet.
